src/SharedProtectWebSocket.ts

```diff
@@ -0,0 +1,145 @@
+import WebSocket from 'ws'
+import AccessControllerNodeConfigType from './types/AccessControllerNodeConfigType'
+import AccessControllerNodeType from './types/AccessControllerNodeType'
+import { endpoints } from './Endpoints'
+import { ProtectApiUpdates } from './lib/ProtectApiUpdates'
+import { logger } from '@nrchkb/logger'
+import { Loggers } from '@nrchkb/logger/src/types'
+import * as crypto from 'crypto'
+
+/**
+ * DEFAULT_RECONNECT_TIMEOUT is to wait until next try to connect web socket in case of error or server side closed socket (for example UniFi restart)
+ */
+const DEFAULT_RECONNECT_TIMEOUT = 90000
+export type WSDataCallback = (data: any) => void
+export interface Interest {
+    deviceId: string
+    callback: WSDataCallback
+}
+
+export class SharedProtectWebSocket {
+    private bootstrap: any
```

Comment:
Cannot we find a type for that?

Reply:
I don't know all possible properties for the bootstrap

Reply:
I think bootstrap either JSON or an error?

Reply:
@Shaquu & @crxporter 
I have moved from  `any` to `Record<string, any>` to identify these objects as JSON objects
this is for the bootstrap and `shapeProfile`.